Support Composable object as sql statement for psycopg2

condorcet opened the pull request:
Fix https://github.com/uber-common/opentracing-python-instrumentation/issues/81

condorcet commented:
I've added `quote_ident` wrapper, because I've found that `sql.Identifier` using `psycopg2.extensions.quote_ident` function to make string representation, but the function accept only cursor / connection objects. Also I've added test cases for each `Composable` classes.

condorcet commented on opentracing_instrumentation/client_hooks/psycopg2.py:
```diff
@@ -31,25 +31,49 @@
+    executing.
+    """
+    def execute(self, sql, params=NO_ARG):
+        if not isinstance(sql, str):
```
Hm, it can be unicode too. So it's better to check is sql instance of `Composable` class.

yurishkuro commented on opentracing_instrumentation/client_hooks/_dbapi2.py:
```diff
@@ -81,6 +81,48 @@ def empty_ctx_mgr():
     )
 
 
+class CursorWrapper(wrapt.ObjectProxy):
```
why move this code?

Jamim replied:
Hi @yurishkuro,

I believe that @condorcet moved it because `CursorWrapper` must be defined before it's used to set `cursor_cls` attribute of `ConnectionWrapper` class.

condorcet replied:
Absolutely right, that's why I moved this code

yurishkuro commented on opentracing_instrumentation/client_hooks/_dbapi2.py:
```diff
@@ -119,13 +161,15 @@ def __call__(self, *args, **kwargs):
 
     def cursor(self, *args, **kwargs):
-        return CursorWrapper(
+        return self.cursor_cls(
```
`cursor_cls` is defined as class field, but you're referencing it as member field via `self.`

Jamim replied:
Hi @yurishkuro,

You have to use `self` to access `cursor_cls` (that might be overridden) since `cursor` is not a `classmethod` and you have no `cls` here.
Surely, you can use `self.__class__.cursor_cls`, but it's completely redundant.

So it's a normal approach, there is nothing bad here actually.

condorcet replied:
Yes, it is a common pattern in python to access a class field through instance of the class.

yurishkuro replied:
A common pattern doesn't mean it's a good pattern.

Why not do it cleanly as a member variable initialized with value passed to the constructor?

condorcet replied:
You are right, I did it with additional argument in `ConnectionFactory` constructor.

condorcet commented:
Everything okay, I deleted my branch by mistake :) In last commit I added casting to string a `Composable` sql object for `executemany` cursor method.
